src/ol/source/UrlTile.js

```diff
@@ -64,6 +64,7 @@ class UrlTile extends TileSource {
      */
     this.tileUrlFunction = this.fixedTileUrlFunction ?
       this.fixedTileUrlFunction.bind(this) : nullTileUrlFunction;
+    this.key_ = options.opt_key || this.key_;
```

Comment:
Since this is a new property for the `UrlTile` constructor options, it needs to be added to the [`@typedef` here](https://github.com/openlayers/openlayers/blob/7d77d05a575ab98391ae9de4038585bd2bfa1bee/src/ol/source/UrlTile.js#L11-L27) as well.  As above, I would change the property name to just `key`.